fuzzer/diff_fuzzer/README.md

```diff
@@ -0,0 +1,61 @@
+# Random hint fuzzer
+
+This fuzzer takes a list of pre-selected hints and runs a generated basic program with randomized inputs that uses a random hint from the list.
+
+
+## To run the fuzzer 
+
+Ensure that you have the atheris module installed before running, if not run `pip install atheris`.
+Use the command `make diff-fuzz` to run the fuzzer.
+
+
+## The fuzzer
+The fuzzer is located in the ***random_hint_fuzzer.py*** file inside the diff_fuzzer folder.
+
+### How does it work?
+
+1. first it selects a random hint with the function `get_random_hint`, it has a list of indexes that represent a hint previously saved in a file within ***/hint_accountant/whitelists/***; Selects a random number between the listed ones and returns the list of lines that correspond to that hint code.
+2. Creates a basic program that uses the hint calling the function `generate_cairo_hint_program`, it returns a program with a `REPLACEABLE_TOKEN` constant placed wherever a random value has to be placed.
+3. Replace all the constants with random values with the function `generate_limb`, this function returns a value with 70% of provability to be a number within ***range_check_max >> 1 and range_check_max***, with range_check_max being ***"340282366920938463463374607431768211456"*** 15% probability of being a number within ***0 and 10*** and 15% probability of being between ***1 and range_check_max***.
+4. Creates a random name for the modified Cairo file and the compiled json one.
+5. Write the modified cairo program to a file with the generated ***.cairo*** name.
+6. Compile the ***.cairo*** file to get the ***.json*** file.
+7. Run the ***.json*** program with the Python original vm implementation and the new Rust one.
+8. if both implementations run correctly, use the `check_mem` function to compare that the memories are the same. If one implementation returns an error and the other one run correctly the fuzzer reports an error, as both implementations should return the same.
```

Comment:
```suggestion
8. if both implementations run correctly, use the `check_mem` function to compare that the memories are the same. If one implementation returns an error and the other one runs correctly the fuzzer reports an error, as both implementations should return the same output.
```

Reply:
Commited the suggestion in: https://github.com/lambdaclass/cairo-vm/pull/1358/commits/66272d837aa276d9f93c1a21222d3ab38d1a9f4a